WalletWasabi.Fluent/Views/AddWallet/Create/ConfirmRecoveryWordsView.axaml

```diff
@@ -12,29 +12,95 @@
              x:CompileBindings="True"
              x:Class="WalletWasabi.Fluent.Views.AddWallet.Create.ConfirmRecoveryWordsView">
   <c:ContentArea Title="{Binding Title}"
+                 Caption="{Binding CurrentWord.Index, StringFormat='Click the recovery word #{0}', FallbackValue=' '}"
                  EnableCancel="{Binding EnableCancel}"
                  EnableBack="{Binding EnableBack}"
-                 EnableNext="True" NextContent="Continue"
-                 EnableSkip="{Binding IsSkipEnable}"
-                 IsBusy="{Binding IsBusy}">
-    <ScrollViewer HorizontalAlignment="Stretch" VerticalAlignment="Center" MaxWidth="350">
-      <ItemsControl Items="{Binding ConfirmationWords}">
-        <i:Interaction.Behaviors>
-          <behaviors:FocusFirstTextBoxInItemsControlBehavior />
-        </i:Interaction.Behaviors>
-        <ItemsControl.ItemTemplate>
-          <DataTemplate>
-            <DockPanel>
-              <Label Content="{Binding Index, StringFormat=Recovery word {0}, Mode=OneWay}" DockPanel.Dock="Top" />
-              <TextBox Watermark="Type in the word" Classes="hasCheckMarkBoolean" validation:CheckMarkStatus.IsEnabled="{Binding IsConfirmed}" Text="{Binding Input}" IsEnabled="{Binding !IsConfirmed}" DockPanel.Dock="Top">
-                <i:Interaction.Behaviors>
-                  <behaviors:FocusNextItemBehavior IsFocused="{Binding !IsConfirmed}" />
-                </i:Interaction.Behaviors>
-              </TextBox>
-            </DockPanel>
+                 EnableNext="True"
+                 NextContent="Continue"
+                 EnableSkip="{Binding IsSkipEnabled}"
+                 IsBusy="{Binding IsBusy}"
+                 ScrollViewer.VerticalScrollBarVisibility="Disabled">
+    <Grid x:Name="Layout">
+      <Grid.RowDefinitions>
+        <RowDefinition MinHeight="70" Height="*" />
+        <RowDefinition Height="16" />
+        <RowDefinition Height="Auto" />
+        <RowDefinition Height="*" />
+        <RowDefinition Height="20" />
+      </Grid.RowDefinitions>
+
+      <c:RecoveryWordsControl Items="{Binding AvailableWords}">
+        <c:RecoveryWordsControl.Styles>
+          <Style Selector="StackPanel">
+            <Setter Property="HorizontalAlignment" Value="Center" />
+          </Style>
+          <Style Selector="ItemsControl">
+            <Setter Property="ItemsPanel">
+              <ItemsPanelTemplate>
+                <StackPanel Orientation="Horizontal" />
+              </ItemsPanelTemplate>
+            </Setter>
+          </Style>
+        </c:RecoveryWordsControl.Styles>
+        <c:RecoveryWordsControl.ItemTemplate>
+          <DataTemplate x:DataType="create:RecoveryWordViewModel">
+            <CheckBox IsChecked="{Binding IsSelected}"
```

Comment:
The use of checkbox here is extremely hacky!

Just bind to IsSelected and use a converter to change the background (we already have boolean converters)

User a border or set the background on TagControl directly.

Reply:
Switched to `ToggleButton`.